cvprac/cvp_api.py

```diff
@@ -3828,3 +3839,202 @@ def device_decommissioning_status_get_all(self, status="DECOMMISSIONING_STATUS_U
+            return self.clnt.post(url)
+
+    def svc_account_get_one(self, username):
+        ''' Get an arbitrary service account's state using Resource APIs
+            Supported versions: CVP 2021.3.0 or newer and CVaaS.
+            Args:
+                username (string): The service account username.
+            Returns:
+                response (list): Returns a list of dict that contains...
+                Ex: [{'value': {'key': {'name': 'ansible'}, 'status': 'ACCOUNT_STATUS_ENABLED',
+                      'description': 'lab-tests', 'groups': {'values': ['network-admin']}},
+                      'time': '2022-02-10T04:28:14.251684869Z'}]
+        '''
+        msg = 'Service Account Resource APIs are supported from 2021.3.0+.'
+        if self.check_v7(msg):
+            payload = {"key": {"name": username}}
+            url = '/api/v3/services/arista.serviceaccount.v1.AccountConfigService/GetOne'
+            self.log.debug('v7 {} {}'.format(url, payload))
+            return self.clnt.post(url, data=payload)
+
+    def svc_account_set(self, username, description, roles, status):
+        ''' Create a service account token using Resource APIs.
+            Supported versions: CVP 2021.3.0 or newer and CVaaS.
+            Args:
+                username (string): The service account username.
+                description (string): The description of the service account.
+                roles (list): The list of role IDs. Default roles have a human readable name,
+                    e.g.: 'network-admin', 'network-operator';
+                    other roles will have the format of 'role_<unix_timestamp>',
+                    e.g. 'role_1658850344592739349'.
+                    cvprac automatically converts non-default role names to role IDs.
+                status (enum): The status of the service account. Possible values:
+                    0 or 'ACCOUNT_STATUS_UNSPECIFIED'
+                    1 or 'ACCOUNT_STATUS_ENABLED'
+                    2 or 'ACCOUNT_STATUS_DISABLED'
+            Returns:
+                response (list): Returns a list of dict that contains...
+                Ex: [{'value': {'key': {'name': 'cvprac2'}, 'status': 'ACCOUNT_STATUS_ENABLED',
+                      'description': 'testapi', 'groups': {'values':
+                      ['network-admin', 'role_1658850344592739349']}},
+                      'time': '2022-07-26T18:19:55.392173445Z'}]
+        '''
+        msg = 'Service Account Resource APIs are supported from 2021.3.0+.'
+        if self.check_v7(msg):
+            role_ids = []
+            all_roles = self.get_roles()
+            for r in roles:
+                for role in all_roles['roles']:
+                    if r == role['key']:
+                        role_ids.append(r)
+                        break
+                    elif r == role['name']:
+                        role_ids.append(role['key'])
+                        break
```

Comment:
I think this can be simplified to something like this:

* I am assuming that the roles provided to the function is a list of string.

```
for role in all_roles['roles']:
    if role['key'] in roles or role['name'] in roles:
        role_ids.append(role['key'])
```

The only thing this doesn't include is if a provided role string in the roles list isn't included. We could add this as well via something like:

```
if len(roles) != len(role_ids):
    self.log.warning('Not all provided roles {} are valid. Only using found valid roles {}'.format(roles, role_ids))
```